lib/Segment/Consumer/LibCurl.php

```diff
@@ -49,21 +49,33 @@ public function flushBatch($messages) {
     curl_setopt($ch, CURLOPT_URL, $url);
     curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
 
-    // execute post
-    curl_exec($ch);
+    // retry failed requests just once to diminish impact on performance
+    $httpResponse = $this->executePost($ch);
 
-    if ($this->debug()) {
+    if ($httpResponse != 200) {
 
-      $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+      // log error
+      $this->handleError($ch, $httpResponse);
 
-      if ($httpCode != 200) {
-        $this->handleError($ch, $httpCode);
-      }
+      //wait 300ms
+      usleep(300000);
```

Comment:
I think we might want to dial this back a bit since 300ms is a long time. I'm guessing that just retrying immediately is enough in most cases so that we don't block the execution, but still have a chance of making a fresh connection.